### PR TITLE
 add `matrix` function to phase out `mat`

### PR DESCRIPTION
**Pull request by felix-roehrich**
- add `matrix` function to phase out `mat` (in later PR otherwise OSCAR CI will fail)
- change occurrences of `mat` to `matrix`

Also see https://github.com/oscar-system/Oscar.jl/issues/2885.

### CODE REVIEW

**Review by fingolfin — APPROVED** [review]

**Comment by felix-roehrich** [comment]
The test against Oscar/master was not run, so this can't be merged. The other matching tests are failing, but they can be skipped on a rerun, as this is independent of this PR (https://github.com/oscar-system/Oscar.jl/actions/runs/6396107746/job/17361226372).

File: src/ModuleHomomorphism.jl [hunk 1]
```diff
@@ -12,6 +12,8 @@
 
 mat(f::Map(FPModuleHomomorphism)) = f.matrix
```
Comment (lgoettgens) [inline src/ModuleHomomorphism.jl:13]: Maybe replace this by a deprecation instead? So we do not forget to remove it sometime in the future?

**Comment by thofma** [comment]
The Oscar error looked suspicious: https://github.com/Nemocas/AbstractAlgebra.jl/actions/runs/6430182583/job/17474628140?pr=1461

I have restarted

**Comment by fingolfin** [comment]
It seems this is a breaking change after all, because of how it interacts with the `mat` method of Oscar's GModule code. It may be possible to avoid this by using `@alias` here instead of `@deprecated` (or perhaps `Base.@deprecate_binding` -- see also https://github.com/JuliaLang/julia/issues/51360)

**Comment by felix-roehrich** [comment]
This is not a breaking change (or shouldn't be). The branch tested against `Oscar/fr-mat-matrix` is based on the `CF-WorkInProgress` branch, which as said in my earlier comment, was already failing. The relevant test would be against `Oscar/master` but this isn't run.

**Comment by fingolfin** [comment]
OK, fine by me then. And in any case, the next AA release I am about to make is already a "breaking" one either way, so we should be fine.

**Pull request merged**